File: Python Solutions/School Round 2017/Round 1/Problem 3 Reading Level Solution 1.py
from string import ascii_lowercase

if __name__ == '__main__':
    sentance = list(input('Input: '))
    for l, letter in enumerate(sentance):
        if letter.lower() not in ascii_lowercase:
            sentance[l] = ' '
    sentance = ''.join(sentance)
    sentance = sentance.replace('  ','')
    if sentance[-1] == ' ':
        sentance = sentance[:-1]
    sentance = sentance.split(' ')
    print(sum(map(lambda word:len(word),sentance))/len(sentance))

'''
“Stop!”, shouted Chris. He said to Emily, “Why were you running in the hall?”
'''

File: C++_Solutions/SAPO/School_Round_2017/Round_1/Problem_3_Reading_Level_Solution_1.cpp
#include <iostream>
#include <cctype>

using namespace std;

string letters = "abcdefghijklmnopqrstuvwxyz";

void remove_symbols(string& word){
  for (size_t i = 0; i < word.size(); i++) {
    if(letters.find(tolower(word[i])) == string::npos){
      word[i] = ' ';
    }
  }
}
int count_words(string& word){
  int counter = 0;
  bool space = true;
  for (size_t i = 0; i < word.size(); i++) {
    if(word[i-1] == ' ' && word[i] != ' '){
      counter++;
    }
  }
  return counter;
}

int count_letters(string& word){
  int counter = 0;
  for (size_t i = 0; i < word.size(); i++) {
    if(letters.find(tolower(word[i])) != string::npos){
        counter++;
    }
  }
  return counter;
}

int main(int argc, char const *argv[]) {

  string word = "\“Stop!\”, shouted Chris. He said to Emily, \“Why were you running in the hall?\”";
  remove_symbols(word);
  cout<<(long double)count_letters(word)/count_words(word)<<endl;
  return 0;
}
